src/hotspot/cpu/x86/c2_MacroAssembler_x86.cpp

```diff
@@ -1045,6 +1045,42 @@ void C2_MacroAssembler::evminmax_fp(int opcode, BasicType elem_bt,
   }
 }
 
+// Float/Double signum
+void C2_MacroAssembler::signum_fp(int opcode, XMMRegister dst,
+                                  XMMRegister zero, XMMRegister one,
+                                  Register scratch) {
+  assert(opcode == Op_SignumF || opcode == Op_SignumD, "sanity");
+
+  Label DONE_LABEL;
+
+  if (opcode == Op_SignumF){
+    assert(UseSSE > 0, "required");
+    ucomiss(dst, zero);
+  } else if (opcode == Op_SignumD){
+    assert(UseSSE > 1, "required");
+    ucomisd(dst, zero);
+  }
+
+  jcc(Assembler::equal, DONE_LABEL);    // handle special case +0.0/-0.0, if argument is +0.0/-0.0, return argument
+  jcc(Assembler::parity, DONE_LABEL);   // handle special case NaN, if argument NaN, return NaN
+
+  if (opcode == Op_SignumF){
```

Comment:
Too many if/else, can you kindly club the instruction for SignumF and SignumD in one conditional(if/else) block.